### ISSUE TITLE
[Donut Chart]: Donut Chart is not keyboard accessible

### ISSUE THREAD

**Issue by amdev138** [UI5/webcomponents-react#7609]
### Describe the bug

Hi team,

I see in recharts storybook, they have a way make the donut charts keyboard accessible. Please see either:
1. [recharts storybook](https://recharts.org/en-US/storybook) > Pie > CustomActiveShapePiechart
2. Alternatively you can navigate directly to the iframe: [custom-active-shape-pie-chart](https://main--63da8268a0da9970db6992aa.chromatic.com/iframe.html?globals=&args=&id=examples-pie-customactiveshapepiechart--custom-active-shape-pie-chart)

However, I could not seem to get this to work through the react wrapper. I suspect it may be due to this rootTabIndex always set to -1, [PieChart.tsx#L316](https://github.com/SAP/ui5-webcomponents-react/blob/main/packages/charts/src/components/PieChart/PieChart.tsx#L316), but I could not verify that. 

Please help to fix or find a solution to make this chart keyboard accessible.

Thanks!

### Reproduction steps

1. Find the storybook links in the description of this ticket, tab to the chart container, then use the arrow keys to traverse the pie chart.
2. Open the [ui5-webcomponents-react storybook](https://sap.github.io/ui5-webcomponents-react/v2/?path=/story/charts-piechart--default) and try the same. Here it will not be possible to traverse the pie chart with the keyboard.

### Expected Behaviour

Developers should be able to leverage the keyboard accessibility that recharts seems to offer.

### UI5 Web Components for React Version

main

### UI5 Web Components Version

main

### Browser

Chrome

### Relevant log output

```shell

```

### Organization

SF

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Hi @amdev138,

Recharts was recently updated to version 3, which introduces the `accessibilityLayer` as a default feature across all charts. This change is not present in v2, which likely explains the difference in behavior you're seeing.

We've already started migrating our charts to v3, but I can't give a definitive timeline for completion yet. We're facing some challenges with custom implementations that are no longer compatible. You can track our progress [here](https://github.com/SAP/ui5-webcomponents-react/pull/7617).

**Comment by amdev138** [comment]
Hi @Lukas742 , any updates on this issue?

**Comment by Lukas742** [comment]
Hi @amdev138 

sorry for the late reply. It turns out that updating to recharts v3 is currently not feasible as it would entail breaking changes. Due to our current setup it is not possible publishing only `@ui5/webcomponents-react-charts`, so we have to stick with `recharts` v2 for now.
That said, the linked PR will add keyboard support to the Pie and Donut chart, when `chartConfig.accessibilityLayer` is active. Before the `accessibilityLayer` was a no-op for Pie and Donut chart, that's why we added support for it manually. 

Please note that there are very limited SAP design or a11y specifications for charts, so the keyboard navigation is our best effort approach based on common patterns.

**Comment by amdev138** [comment]
@Lukas742 , I understand. Thanks for the support!
